service/src/main/kotlin/fi/espoo/evaka/invoicing/service/InvoiceCorrectionQueries.kt

```diff
@@ -0,0 +1,157 @@
+// SPDX-FileCopyrightText: 2017-2024 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.invoicing.service
+
+import fi.espoo.evaka.shared.ChildId
+import fi.espoo.evaka.shared.DaycareId
+import fi.espoo.evaka.shared.InvoiceCorrectionId
+import fi.espoo.evaka.shared.PersonId
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.db.Predicate
+import fi.espoo.evaka.shared.db.PredicateSql
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import java.time.LocalDate
+import java.time.YearMonth
+
+private fun Database.Read.getInvoiceCorrections(where: Predicate) =
+    createQuery {
+            sql(
+                """
+SELECT
+    id,
+    target_month,
+    head_of_family_id,
+    child_id,
+    unit_id,
+    product,
+    period,
+    amount,
+    unit_price,
+    description,
+    note
+FROM invoice_correction
+WHERE
+    target_month IS NOT NULL AND
+    ${predicate(where.forTable("invoice_correction"))}
+"""
+            )
+        }
+        .toList {
+            InvoiceCorrection(
+                id = column("id"),
+                targetMonth = YearMonth.from(column<LocalDate>("target_month")),
```

Comment:
maybe we could consider creating a postgres type for year_month with date as internal representation, which would then automatically serialize correctly